Sources/NIO/Bootstrap.swift

```diff
@@ -967,3 +979,44 @@ public final class NIOPipeBootstrap {
         }
     }
 }
+
+/// An channel option which can be applied to bootstrap using shorthand notation.
+/// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+public struct ShorthandBootstrapOption {
+    private let apply : ShorthandApply
+    
+    private init(apply : ShorthandApply) {
+        self.apply = apply
+    }
+    
+    /// Apply the contained option to the supplied ServerBootstrap
+    /// - Parameter serverBootstrap: bootstrap to apply this option to.
+    /// - Returns:the modified bootstrap (currently the same one mutated)
+    func applyOption<BootstrapT : ServerBootstrap>(serverBootstrap : BootstrapT) -> BootstrapT {
+        return apply.applyOption(serverBootstrap: serverBootstrap)
+    }
+}
+
+private protocol ShorthandApply {
+    func applyOption<BootstrapT : ServerBootstrap>(serverBootstrap : BootstrapT) -> BootstrapT
+}
+
+private struct ShorthandApplyImpl<Option : ChannelOption> : ShorthandApply {
+    let option : Option
+    let value : Option.Value
+    
+    func applyOption<BootstrapT : ServerBootstrap>(serverBootstrap: BootstrapT) -> BootstrapT {
+        return serverBootstrap.serverChannelOption(option, value: value)
+    }
+}
```

Comment:
I think the protocol + implementation abstraction here is not helpful: we don't expect these types to ever be generic or for there to ever be more than one implementation, so let's just use the concrete type directly.

Reply:
I make this generic because the called serverChannelOption is generic.
If I keep it generic but remove the protocol  then ShortHandBootstrap becomes generic which means I can't extend it so the .reuseAddr syntax doesn't work any more.
Are you suggesting I make everything concrete on ChannelOption itself?

Reply:
`serverChannelOption` is generic, but not over the bootstrap. It's generic over the `Option`. And that's not generic here: we know what the option type is concretely.